### PR TITLE
Link company to users

### PR DESCRIPTION
**Pull request by JoseRicardoJimenez**
# Description
What did you implemented/Why did you implemented this:
 - i've linked company to users, modifing model and creating a new migration
 - i've updated business veiws

 
 Examples:
 - What?: I've added support for authentication to implement Key Result 2 of OKR1. It includes 
model, table, controller and test. For more background, see ticket #JIRA-123.
 - Why?: These changes complete the user login and account creation experience. See #JIRA-123 for more information.
 
 # Testing
 Help me how can I test or look at the changes
 
 Example:
  - I've added coverage for testing all new methods. I used Faker for a few random user emails and names.
  
  ## Screenshots 
If applicable, include screenshots of the results or screenshots that help to see changes

### CODE REVIEW

**Review by YilKanda — APPROVED** [review]
LGTM!

File: app/models/company.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class Company < ApplicationRecord
- validates :name, :address, :phone, :category_id, :town_id, presence: true
+ validates :name, :address, :phone, :category_id, :town_id, :user_id, presence: true
 
+  belongs_to :user
```
Comment (dannharo) [inline app/models/company.rb:4]: I think you also need to add the relation in the [User](https://github.com/BC-MAY-21-ROR/proyecto-proy-t02/blob/master/app/models/user.rb#L1) controller...

File: app/views/administrators/bussiness.html.erb [hunk 2]
```diff
@@ -7,16 +7,16 @@
...
       </thead>
       <tbody>
         <% @companies.each do |company| %>
           <tr>
-            <td class="py-4"><%= company.name %></td>           
-            <td class="py-4"></td>
+            <td class="py-4"><%= company.name %></td> 
+            <td class="py-4"><%= @companies.find(company.id).user.email %></td>
             <td class="py-4 px-1"><%= @companies.find(company.id).category.name %></td>
             <td class="py-4 px-1"><%= Product.where(company_id: company.id).sum(:quantity)%> </td>
```
Comment (dannharo) [inline app/views/administrators/bussiness.html.erb:21]: avoid to add logic in your views, this should only show the data passed through the controller...

File: app/views/administrators/bussiness.html.erb [hunk 3]
```diff
@@ -7,16 +7,16 @@
...
           <th class="w-28"> Productos Totales </th>
         </tr>
       </thead>
       <tbody>
         <% @companies.each do |company| %>
           <tr>
-            <td class="py-4"><%= company.name %></td>           
-            <td class="py-4"></td>
+            <td class="py-4"><%= company.name %></td> 
+            <td class="py-4"><%= @companies.find(company.id).user.email %></td>
```
Comment (dannharo) [inline app/views/administrators/bussiness.html.erb:19]: I think you don't need to do the `find` here if the relation is `ok` you are going to be able to do something like this:
`company.user.email.`

**Review by dannharo — COMMENTED** [review]
Add tests, at least in the Models. 
Add screenshots for the changes in the `View`

File: app/models/company.rb (hunk 1 above)
Comment (oramasangel) [reply app/models/company.rb:4]: we'll add it

**Review by oramasangel — APPROVED** [review]

**Review by jairolan — DISMISSED** [review]

File: app/views/administrators/bussiness.html.erb (hunk 2 above)
Comment (YilKanda) [reply app/views/administrators/bussiness.html.erb:21]: Please pay attention to @dannharo 's feedback, since this logic should not appear in their views.

**Review by YilKanda — CHANGES_REQUESTED** [review]
Do not put syntax corresponding to the controllers in the views, because it can generate internal security flaws on your website.

**Review by brayandiazc — DISMISSED** [review]

**Review by oramasangel — APPROVED** [review]

File: app/models/company.rb [hunk 4]
```diff
@@ -1,10 +1,15 @@
...
   belongs_to :category
   belongs_to :town
   has_many :providers
   has_many :products
   has_one_attached :company_img
-  
+
+  def sum_products()
+    products.sum(:quantity)
+  end
```
Comment (juliocabrera820) [inline app/models/company.rb:13]: A name more descriptive would be *products_total* and remove the parentheses

**Review by juliocabrera820 — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]
LGTM!

**Pull request merged**